File: 양승민/입문/src/lec07/lec07Main.java
package lec07;

import org.jetbrains.annotations.NotNull;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class lec07Main {
    public static void main(String[] args) {

    }

    // 1. try catch finally
    private int parseIntOrThrow(@NotNull String string) {
        try {
            return Integer.parseInt(string);
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException(e);
        }
    }

    private Integer parseIntOrThrow2(@NotNull String string) {
        try {
            return Integer.parseInt(string);
        } catch (NumberFormatException e) {
            return null;
        }
    }

    // 3. try with resources
    public void readFile(String path) throws IOException {
        try (BufferedReader reader = new BufferedReader(new FileReader(path))) {
            System.out.println(reader.readLine());
        }
    }

}

File: 양승민/입문/src/lec07/lec07Main.kt
package lec07

import java.io.BufferedReader
import java.io.FileReader

fun main() {

    // 1. try catch finally
    fun parseIntOrThrow(string: String): Int {
        try{
            return string.toInt()
        }catch (e: NumberFormatException){
            throw IllegalArgumentException(e)
        }
    }

    fun parseIntOrThrow2(string: String): Int? {
        return try{
            return string.toInt()
        }catch (e: NumberFormatException){
            null;
        }
    }

    // 3. try with resources
    fun readFile(path: String) {
        BufferedReader(FileReader(path)).use { reader ->
            println(reader.readLine())
        }
    }

}